hpcflow/sdk/helper/helper.py

```diff
@@ -255,14 +334,40 @@ def run_helper(
         timeout_check_interval = timedelta(seconds=timeout_check_interval_s)
 
     start_time = datetime.now()
+    end_time = start_time + timeout
     logger = get_helper_logger(app)
     controller = MonitorController(get_watcher_file_path(app), watch_interval, logger)
-    timeout_limit = timeout - timeout_check_interval
+    helper_args = read_helper_args(app)
     try:
         while True:
-            if datetime.now() - start_time >= timeout_limit:
+            time_left_s = (end_time - datetime.now()).total_seconds()
+            if time_left_s <= 0:
                 helper_timeout(app, timeout, controller, logger)
-            time.sleep(timeout_check_interval_s)
+            time.sleep(min(timeout_check_interval_s, time_left_s))
+            # Reading args from PID file
+            helper_args_new = read_helper_args(app)
+            for arg in helper_args:
+                if helper_args_new[arg] != helper_args[arg]:
+                    change = f"{arg} parameter from {helper_args[arg]} to {helper_args_new[arg]}."
+                    helper_args[arg] = helper_args_new[arg]
+                    t = helper_args[arg]
+                    if isinstance(t, timedelta):
+                        t_s = t.total_seconds()
+                    else:
+                        t_s = t
+                        t = timedelta(seconds=t_s)
+                    # Updates relevant parameters only
+                    if arg == "timeout":
+                        timeout = t
+                        end_time = start_time + timeout
+                    elif arg == "timeout_check_interval":
+                        timeout_check_interval_s = t_s
+                    elif arg == "watch_interval":
+                        controller.stop()
```

Comment:
Could you add the TODO note back in here, regarding missing a workflow change?

Reply:
Resolved in [e0569e4](https://github.com/hpcflow/hpcflow-new/pull/340/commits/e0569e4048b9228aea4504b4bb61e6e73569b5ef)